packages/@aws-cdk/aws-cloudwatch/lib/graph.ts

```diff
@@ -364,6 +364,97 @@ export class SingleValueWidget extends ConcreteWidget {
   }
 }
 
+/**
+ * The properties for a CustomWidget
+ */
+export interface CustomWidgetProps {
+  /**
+   * The Arn of the AWS Lambda function that returns HTML or JSON that will be displayed in the widget
+   */
+  readonly functionArn: string;
+
+  /**
+   * Width of the widget, in a grid of 24 units wide
+   *
+   * @default 6
+   */
+  readonly width?: number;
+
+  /**
+   * Height of the widget
+   *
+   * @default - 6 for Alarm and Graph widgets.
+   *   3 for single value widgets where most recent value of a metric is displayed.
+   */
+  readonly height?: number;
+
+  /**
+   * The title of the widget
+   */
+  readonly title: string;
+
+  /**
+   * Update the widget on refresh
+   *
+   * @default true
+   */
+  readonly updateOnRefresh?: boolean;
+
+  /**
+   * Update the widget on resize
+   *
+   * @default true
+   */
+  readonly updateOnResize?: boolean;
+
+  /**
+   * Update the widget on time range change
+   *
+   * @default true
+   */
+  readonly updateOnTimeRangeChange?: boolean;
+
+  /**
+   * Parameters passed to the lambda function
+   *
+   * @default {}
+   */
+  readonly params?: any;
+}
+
+/**
+ * A CustomWidget shows the result of a AWS lambda function
+ */
+export class CustomWidget extends ConcreteWidget {
+
+  private readonly props: CustomWidgetProps;
+
+  public constructor(props: CustomWidgetProps) {
+    super(props.width ?? 6, props.height ?? 6);
+    this.props = props;
+  }
+
+  public toJson(): any[] {
+    return [{
+      type: 'custom',
+      width: this.width,
+      height: this.height,
+      x: this.x,
+      y: this.y,
+      properties: {
+        endpoint: this.props.functionArn,
+        params: this.props.params ?? {},
```

Comment:
Do you know what the default value of this is if we leave it out? Typically the pattern with CDK is to just leave out properties for the default case if they are not required, and let Cloudformation to handle the default.

Unfortunately, I can't find any documentation for this on the CW documentation. The [Dashboard Body Structure and Syntax](https://docs.aws.amazon.com/AmazonCloudWatch/latest/APIReference/CloudWatch-Dashboard-Body-Structure.html) page is missing the custom widget.

Reply:
As this is only stringified json for cloudformation, I doub't there is a default behaviour. I think you can pass even not working dashboard configs to cloudformation and it will deploy it. So I looked what are the default properties that the web console would set. And this is the result of my next - next - finish test:

```
{
            "type": "custom",
            "x": 0,
            "y": 0,
            "width": 6,
            "height": 6,
            "properties": {
                "endpoint": "arn:aws:lambda:us-east-1:12345678910:function:customWidgetEcho-js",
                "updateOn": {
                    "refresh": true,
                    "resize": true,
                    "timeRange": true
                }
            }
        }
```

Reply:
So what do you think of leaving refresh and resize and timerange to default to true (as the webconsole does) and leave out parameters.

Reply:
This also makes the handling of the updateOn props easier. No check if one of the nested props is set in order to remove the whole updateOn props key ... But if you want I can remove it

Reply:
So for now, i removed params and left the updateOn props. Please tell me your opinion @madeline-k

Reply:
I think it's fine to remove params, and set all of the `updateOn` to true by default. Are you going to make the change to remove params? Right now, it is passing an empty object by default. This is not the same as leaving it out and letting cfn figure it out.

Reply:
:-( I already did that and just forgot to push ...